### ISSUE TITLE
Remove AbstractChannel#doPreClose()

### ISSUE THREAD

**Issue by jpinner** [netty/netty#1654]
tl;dr this is broken and should be removed instead of fixed

doPreClose is currently called in closeFuture.setClosed():

1) During registration failure closeForcily() calls doClose before setting the closeFuture -- so doPreClose() is called after doClose().

2) During early registration failure (unable to schedule run loop) closeFuture is never set so doPreClose() is never called.

3) Exceptions in doPreClose are logged but the caller's promise is not notified.

4) doPreClose is only implemented in the local channels and that logic can be moved to close.

5) close() uses the closeFuture to determine if the socket is closed (via closeFuture.trySuccess) which calls out to user code listening to the closeFuture BEFORE closing the socket!

We should do the following:

1) Move doPreClose to doClose for local sockets.
2) Remove doPreClose and have setClosed() only call trySuccces
3) don't use closeFuture to detect if the socket is closed (use !isOpen) and call closeFuture.setClosed only after calling doClose (which now we can do because we don't have to worry about the doPreClose ordering).
